src/components/organisms/NavBar/UserAuthBtn/UserAuthBtn.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { NavLink, useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import firebase from '../../../../firebase/firebase';
+
+function UserAuthBtn() {
+  const [userLogin, setUserLogin] = useState('');
+  const history = useHistory();
+  const user = useSelector((state) => state.user);
+
+  const loginBtn = useCallback(
+    () => {
+      if (user) {
+        setUserLogin(<span data-test="text-logout">logout</span>);
```

Comment:
because this callback is also used in the useEffect hook I'll keep the logic where it is. The conditional render wouldn't also solve the useEffect render. So best keep things as they are in this instance I think.